chain-impl-mockchain/src/fee.rs

```diff
@@ -45,13 +85,20 @@ impl FeeAlgorithm for LinearFee {
     fn baseline(&self) -> Value {
         Value(self.constant)
     }
+
     fn fees_for_inputs_outputs(&self, inputs: u8, outputs: u8) -> Value {
         Value(
             self.coefficient
                 .saturating_mul((inputs as u64) + (outputs as u64)),
         )
     }
-    fn fees_for_certificate<'a>(&self, _: CertificateSlice<'a>) -> Value {
+
+    fn fees_for_certificate<'a>(&self, cert_slice: CertificateSlice<'a>) -> Value {
+        if let Some(per_certificate_fees) = self.per_certificate_fees {
+            if let Some(fee) = per_certificate_fees.fees_for_certificate(cert_slice) {
+                return fee;
+            }
+        }
```

Comment:
it does not seem to be right to me. @vincenthz said: https://github.com/input-output-hk/chain-libs/pull/168#issuecomment-558119527

> keep the certificate fee and override some values (simplified).

Here you override the whole certificate fee if this value is set. While actually what was asked is to override some of the values. Not necessarily all of them.

i.e. We may still want to have a `LinearFees::certificate` set to a value for all. Except for the stake pool registration certificate where we want to have a different value.

Reply:
> Here you override the whole certificate fee if this value is set

No. There are just 3 types of certificate in `PerCertificateFee`. Probably, it would be a good idea to make every field of `PerCertificateFee` optional.

Note that `PerCertificateFee::fees_for_certificate` may return `None`, in that case we just fall back to `LinearFees::certificate`.